--- a/arch/arm64/boot/dts/rockchip/rk3399-opp.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3399-opp.dtsi
@@ -67,7 +67,7 @@
 			148501   152000   2
 			152001   999999   3
 		>;
-		rockchip,pvtm-freq = <408000>;
+		rockchip,pvtm-freq = <216000>;
 		rockchip,pvtm-volt = <1000000>;
 		rockchip,pvtm-ch = <0 0>;
 		rockchip,pvtm-sample-time = <1000>;
@@ -77,6 +77,15 @@
 		rockchip,pvtm-temp-prop = <115 66>;
 		rockchip,thermal-zone = "soc-thermal";

+		opp-216000000 {
+			opp-hz = /bits/ 64 <216000000>;
+			opp-microvolt = <800000 800000 1250000>;
+			opp-microvolt-L0 = <800000 800000 1250000>;
+			opp-microvolt-L1 = <800000 800000 1250000>;
+			opp-microvolt-L2 = <800000 800000 1250000>;
+			opp-microvolt-L3 = <800000 800000 1250000>;
+			clock-latency-ns = <40000>;
+		};
 		opp-408000000 {
 			opp-hz = /bits/ 64 <408000000>;
 			opp-microvolt = <800000 800000 1250000>;
@@ -132,6 +141,15 @@
 			opp-microvolt-L3 = <1050000 1050000 1250000>;
 			clock-latency-ns = <40000>;
 		};
+		opp-1608000000 {
+			opp-hz = /bits/ 64 <1608000000>;
+			opp-microvolt = <1200000 1200000 1250000>;
+			opp-microvolt-L0 = <1200000 1200000 1250000>;
+			opp-microvolt-L1 = <1175000 1175000 1175000>;
+			opp-microvolt-L2 = <1150000 1150000 1150000>;
+			opp-microvolt-L3 = <1125000 1125000 1125000>;
+			clock-latency-ns = <40000>;
+		};
 	};

 	cluster1_opp: opp-table1 {
@@ -158,7 +176,7 @@
 			155001   160000   2
 			160001   999999   3
 		>;
-		rockchip,pvtm-freq = <408000>;
+		rockchip,pvtm-freq = <216000>;
 		rockchip,pvtm-volt = <1000000>;
 		rockchip,pvtm-ch = <1 0>;
 		rockchip,pvtm-sample-time = <1000>;
@@ -168,6 +186,15 @@
 		rockchip,pvtm-temp-prop = <71 35>;
 		rockchip,thermal-zone = "soc-thermal";

+		opp-216000000 {
+			opp-hz = /bits/ 64 <216000000>;
+			opp-microvolt = <800000 800000 1250000>;
+			opp-microvolt-L0 = <800000 800000 1250000>;
+			opp-microvolt-L1 = <800000 800000 1250000>;
+			opp-microvolt-L2 = <800000 800000 1250000>;
+			opp-microvolt-L3 = <800000 800000 1250000>;
+			clock-latency-ns = <40000>;
+		};
 		opp-408000000 {
 			opp-hz = /bits/ 64 <408000000>;
 			opp-microvolt = <800000 800000 1250000>;
@@ -241,6 +268,15 @@
 			opp-microvolt-L3 = <1125000 1125000 1250000>;
 			clock-latency-ns = <40000>;
 		};
+		opp-2016000000 {
+			opp-hz = /bits/ 64 <2016000000>;
+			opp-microvolt = <1300000 1300000 1300000>;
+			opp-microvolt-L0 = <1300000 1300000 1300000>;
+			opp-microvolt-L1 = <1275000 1275000 1275000>;
+			opp-microvolt-L2 = <1250000 1250000 1250000>;
+			opp-microvolt-L3 = <1225000 1225000 1225000>;
+			clock-latency-ns = <40000>;
+		};
 	};

 	gpu_opp_table: opp-table2 {
